src/main/java/org/broadinstitute/consent/http/db/mapper/StudyReducer.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.consent.http.db.mapper;
+
+import org.broadinstitute.consent.http.enumeration.PropertyType;
+import org.broadinstitute.consent.http.models.FileStorageObject;
+import org.broadinstitute.consent.http.models.Study;
+import org.broadinstitute.consent.http.models.StudyProperty;
+import org.jdbi.v3.core.result.LinkedHashMapRowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class StudyReducer implements LinkedHashMapRowReducer<Integer, Study>, RowMapperHelper {
+
+  @Override
+  public void accumulate(Map<Integer, Study> map, RowView rowView) {
+    Study study =
+            map.computeIfAbsent(
+                    rowView.getColumn("study_id", Integer.class), id -> rowView.getRow(Study.class));
+
+    reduceStudy(study, rowView);
+  }
+
+
+  public void reduceStudy(Study study, RowView rowView) {
+
+    if (hasColumn(rowView, "s_dataset_id", Integer.class)) {
+      study.addDatasetId(rowView.getColumn("s_dataset_id", Integer.class));
+    }
+
+    if (hasColumn(rowView, "sp_study_property_id", Integer.class)) {
+      Integer studyPropertyId = rowView.getColumn("sp_study_property_id", Integer.class);
+      String keyName = rowView.getColumn("sp_key", String.class);
+      String propVal = rowView.getColumn("sp_value", String.class);
+      Integer studyId = rowView.getColumn("sp_study_id", Integer.class);
+      PropertyType propType = PropertyType.String;
+      if (hasColumn(rowView, "sp_type", String.class)) {
+        propType = PropertyType.parse(rowView.getColumn("sp_type", String.class));
+      }
+
+      if (Objects.nonNull(keyName) && Objects.nonNull(propVal)) {
+        try {
+          StudyProperty prop = new StudyProperty();
+          prop.setStudyPropertyId(studyPropertyId);
+          prop.setStudyId(studyId);
+          prop.setStudyId(study.getStudyId());
+          prop.setValue(propType.coerce(propVal));
+          prop.setName(keyName);
+          prop.setType(propType);
+
+          study.addProperty(prop);
+        } catch (Exception e) {
+          // do nothing.
+        }
+      }
+    }
+
+    if (hasColumn(rowView, "fso_file_storage_object_id", Integer.class)
+            && Objects.nonNull(rowView.getColumn("fso_file_storage_object_id", Integer.class))
+    ) {
+      FileStorageObject fileStorageObject = rowView.getRow(FileStorageObject.class);
+
+      switch (fileStorageObject.getCategory()) {
+        case ALTERNATIVE_DATA_SHARING_PLAN -> {
+          if (isFileNewer(fileStorageObject, study.getAlternativeDataSharingPlan())) {
+            study.setAlternativeDataSharingPlan(fileStorageObject);
+          }
+        }
```

Comment:
Right now, datasets also have an alternative data sharing plan. However, that is actively used in the create endpoint - in the following ticket, then it should be removed from datasets alongside changing the FSO creation mechanism.